docker/resource_docker_plugin_funcs.go

```diff
@@ -0,0 +1,165 @@
+	d.Set("disabled", !plugin.Enabled)
+	// TODO support other settings
+	// https://docs.docker.com/engine/reference/commandline/plugin_set/#extended-description
+	// source of mounts .Settings.Mounts
+	// path of devices .Settings.Devices
+	// args .Settings.Args
+	d.Set("args", plugin.Settings.Env)
+}
+
+func resourceDockerPluginRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).DockerClient
+	ctx := context.Background()
+	pluginID := d.Id()
+	plugin, _, err := client.PluginInspectWithRaw(ctx, pluginID)
+	if err != nil {
+		log.Printf("[DEBUG] Inspect a Docker plugin "+pluginID+": %w", err)
+		d.SetId("")
+		return nil
+	}
+	setDockerPlugin(d, plugin)
+	return nil
+}
+
+func setPluginArgs(ctx context.Context, d *schema.ResourceData, client *client.Client, disabled bool) (gErr error) {
+	if !d.HasChange("args") {
+		return nil
+	}
+	pluginID := d.Id()
+	if disabled {
+		// temporarily disable the plugin before updating the plugin settings
+		log.Printf("[DEBUG] Disable a Docker plugin " + pluginID + " temporarily before updating teh pugin settings")
+		if err := client.PluginDisable(ctx, pluginID, types.PluginDisableOptions{
+			Force: d.Get("force_disable").(bool),
+		}); err != nil {
+			return fmt.Errorf("disable the Docker plugin "+pluginID+": %w", err)
+		}
+		defer func() {
+			log.Print("[DEBUG] Enable a Docker plugin " + pluginID)
+			if err := client.PluginEnable(ctx, pluginID, types.PluginEnableOptions{
+				Timeout: d.Get("enable_timeout").(int),
+			}); err != nil {
+				if gErr == nil {
+					gErr = fmt.Errorf("enable the Docker plugin "+pluginID+": %w", err)
+					return
+				}
+				log.Printf("[ERROR] enable the Docker plugin "+pluginID+": %w", err)
+			}
+		}()
+	}
+	log.Printf("[DEBUG] Update settings of a Docker plugin " + pluginID)
+	if err := client.PluginSet(ctx, pluginID, getDockerPluginArgs(d.Get("args"))); err != nil {
+		return fmt.Errorf("modifiy settings for the Docker plugin "+pluginID+": %w", err)
+	}
+	return nil
+}
+
+func resourceDockerPluginUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).DockerClient
+	ctx := context.Background()
+	pluginID := d.Id()
+	skipArgs := false
+	if d.HasChange("disabled") {
+		if d.Get("disabled").(bool) {
+			log.Printf("[DEBUG] Disable a Docker plugin " + pluginID)
+			if err := client.PluginDisable(ctx, pluginID, types.PluginDisableOptions{
+				Force: d.Get("force_disable").(bool),
+			}); err != nil {
+				return fmt.Errorf("disable the Docker plugin "+pluginID+": %w", err)
+			}
+		} else {
+			if err := setPluginArgs(ctx, d, client, false); err != nil {
+				return err
+			}
+			skipArgs = true
+			log.Printf("[DEBUG] Enable a Docker plugin " + pluginID)
+			if err := client.PluginEnable(ctx, pluginID, types.PluginEnableOptions{
+				Timeout: d.Get("enable_timeout").(int),
+			}); err != nil {
+				return fmt.Errorf("enable the Docker plugin "+pluginID+": %w", err)
+			}
+		}
+	}
+	if !skipArgs {
```

Comment:
Can you replace `skipArgs` with ```d.HasChange("disabled") && !d.Get("disabled").(bool)``` to make it more clear what condition the code is executed under?

Reply:
https://github.com/kreuzwerker/terraform-provider-docker/blob/7a79c28c88fd51de0cdb2edd843c63da55ab047e/docker/resource_docker_plugin_funcs.go#L131

skipArgs is used to avoid the unneeded changes.
But the current code is a little complicated and we may be able to refactor.

Reply:
skipArgs == d.HasChange("disabled") && !d.Get("disabled").(bool) when it is evaluated.

Reply:
Oh, I see.
Fixed. 3ec71c5bc968af149cd565aba0e145d14234eaa7